bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,274 @@
+            configurationLock.unlock();
+        }
+
+        try {
+            result = HttpUtil.executeUrl("GET", query, 750);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to query P1 Meter");
+            return;
+        }
+
+        if (result.trim().isEmpty()) {
+            logger.warn("P1 Wi-Fi meter API at URI {} returned empty result", apiURL);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "P1 Meter API returned empty status");
+            return;
+        }
+
+        P1Payload payload = gson.fromJson(result, P1Payload.class);
+        if (payload != null) {
+            if (payload.getMeter_model() == "") {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Results from API seem empty");
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+
+            if (meterModel != payload.getMeter_model()) {
+                meterModel = payload.getMeter_model();
+                updateProperty(HomeWizardBindingConstants.PROPERTY_METER_MODEL, meterModel);
+            }
+
+            if (meterVersion != payload.getSmr_version()) {
+                meterVersion = payload.getSmr_version();
+                updateProperty(HomeWizardBindingConstants.PROPERTY_METER_VERSION, String.format("%d", meterVersion));
+            }
+
+            updateState(HomeWizardBindingConstants.CHANNEL_POWER_IMPORT_T1,
+                    new QuantityType<>(payload.getTotal_power_import_t1_kwh(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_POWER_IMPORT_T2,
+                    new QuantityType<>(payload.getTotal_power_import_t2_kwh(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_POWER_EXPORT_T1,
+                    new QuantityType<>(payload.getTotal_power_export_t1_kwh(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_POWER_EXPORT_T2,
+                    new QuantityType<>(payload.getTotal_power_export_t2_kwh(), Units.WATT));
+
+            updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER,
+                    new QuantityType<>(payload.getActive_power_w(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER_L1,
+                    new QuantityType<>(payload.getActive_power_l1_w(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER_L2,
+                    new QuantityType<>(payload.getActive_power_l2_w(), Units.WATT));
+            updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER_L3,
+                    new QuantityType<>(payload.getActive_power_l3_w(), Units.WATT));
+
+            updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
+                    new QuantityType<>(payload.getTotal_gas_m3(), Units.ONE)); // could convert, 1m^3 = 1000 liters

```

Comment:
`Units.ONE` is not right here. It should probably be:
```suggestion
                    new QuantityType<>(payload.getTotal_gas_m3(), SIUnits.CUBIC_METRE)); // could convert, 1m^3 = 1000 liters

```